crates/lsp/src/lsp.rs

```diff
@@ -332,11 +332,33 @@ impl LanguageServer {
             };
 
             let header = std::str::from_utf8(&buffer)?;
-            let message_len: usize = header
+            let mut segments = header.lines();
+
+            let message_len: usize = segments
+                .next()
+                .ok_or_else(|| anyhow!("unable to find the first line of the LSP message header"))? // Impossible, but just in case.
```

Comment:
Let's remove `ok_or_else|| anyhow!(` with `.context("` calls here and below.
`return Err(anyhow!` could be `anyhow::bail!` too, for readability.